op-node/rollup/driver/driver.go

```diff
@@ -188,7 +189,13 @@ func NewDriver(
 	sys.Register("l1-blocks", l1Tracker, opts)
 
 	l1 = NewMeteredL1Fetcher(l1Tracker, metrics)
-	verifConfDepth := confdepth.NewConfDepth(driverCfg.VerifierConfDepth, statusTracker.L1Head, l1)
+
+	var verifL1 derive.L1Fetcher
+	if driverCfg.UseFinalized {
```

Comment:
i believe we are mixing 2 differents things and that confuses me.

1. Make sure when creating a new block, you don't depend on L1 blocks that are not final (l1 origin). This is an alternative to `SequencerConfDepth`

2. Make sure that when deriving the 'safe' chain we don't derive information from L1 blocks that are not final. But, OP has already a chain pointer for that, right? There's unsafe (not on l1), safe (on eth, but not final), finalize (on eth and final there)

So i don't think we should be mixing those. If we do, the "safe" chain will be 15 (+25 from batcher) minutes delayed from the unsafe chain.

Reply:
I discussed this with Karl a bit more and we agree with Mariano here.
This should only change the sequencer's l1 origin selection. The derivation should use the existing mechanism, l2 blocks get marked as "safe" or "final" when their origin changes its state. My initial idea to avoid reorgs there too will break expectations of users, who couldn't select to see and work on "unsafe" blocks if they trust the sequencer.